### ISSUE TITLE
Add BPE dropout to tokenizers

### ISSUE THREAD

**Issue by c00k1ez** [huggingface/transformers#4039]
# 🚀 Feature request

Hi! 
Thank you for the great library! 
Several days ago I was trying to fine-tune the GPT2 model with a small dataset. It was a bit hard because the model overfitted every time.
So, after it, I found a great paper with simple sub-word regularization, that outperformed classic BPE on text generation tasks (like NMT).
[Original paper](https://arxiv.org/pdf/1910.13267.pdf).
Unfortunately, I can't find any python implementation, only C++ [here](https://github.com/VKCOM/YouTokenToMe/blob/master/youtokentome/cpp/bpe.cpp).

Metrics from the paper:
![pic](https://drive.google.com/uc?export=view&id=1DPbGxppTBj0GG7nfrxmCx2CZxtyHap0v)
